modules/livestore/kafka_client_franz.go

```diff
@@ -0,0 +1,71 @@
+package livestore
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/tempo/pkg/ingest"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/plugin/kprom"
+)
+
+// FranzKafkaClient wraps a kgo.Client and admin client to implement the KafkaClient interface
+type FranzKafkaClient struct {
+	client  *kgo.Client
+	adminCl *kadm.Client
+}
+
+// NewFranzKafkaClient creates a new Franz/kgo-based Kafka client
+func NewFranzKafkaClient(cfg ingest.KafkaConfig, metrics *kprom.Metrics, logger log.Logger) (KafkaClient, error) {
+	// Create client with topic consumption configured
+	opts := []kgo.Opt{
```

Comment:
why are we adding these? the ingest.NewReaderClient was called with default opts before?

Reply:
It doesnt need it here removed.